### ISSUE TITLE
MetricsPanelCtrl incompletely declared in common.d.ts

### ISSUE THREAD

**Issue by ianwarnold** [grafana/grafana-sdk-mocks#2]
I have effectively no prior experience with TypeScript (or JavaScript for that matter), so feel free to disregard this if it's ridiculous.

I'm looking at a panel developed internally at my organization that was originally copied from a builtin Grafana panel. I'm working on converting it to a proper plugin.

The controller for this panel extends `MetricsPanelCtrl`, which it imports from `app/plugins/sdk`. When built in the Grafana source tree, as it currently stands, this import ultimately resolves to the real `MetricsPanelCtrl`. As a plugin, using grafana-sdk-mocks with a `reference path` directive to include `app/headers/common.d.ts`, it resolves to this empty definition:

https://github.com/grafana/grafana-sdk-mocks/blob/4eead586bdba3f5030db894aeb8a316fc853dad7/app/headers/common.d.ts#L161

I therefore get a bunch of errors from the TypeScript compiler related to fields and methods not being found.

Should the definition linked above perhaps be fleshed out to include declarations to extend `PanelCtrl`, and to contain the fields and methods that the real `MetricsPanelCtrl` contains?

**Comment by hagen1778** [comment]
I'm admit that I am dumb at js/ts but here's my solution of this issue #3

**Comment by daniellee** [comment]
This lib is a work in progress - I created this lib when working on the [Azure Monitor data source](https://github.com/grafana/azure-monitor-datasource) which is a data source plugin. I have plans to convert the Worldmap and Clock plugins to TypeScript and to using this lib as well but I haven't got that far yet.

I just merged @hagen1778 's PR which should hopefully solve this for you. Let me know if anything else is missing and I will add it.
